lib/rack/utils.rb

```diff
@@ -248,13 +248,18 @@ def add_cookie_to_header(header, key, value)
           rfc2822(value[:expires].clone.gmtime) if value[:expires]
         secure = "; secure"  if value[:secure]
         httponly = "; HttpOnly" if (value.key?(:httponly) ? value[:httponly] : value[:http_only])
-        first_party = "; First-Party" if value[:first_party]
+        same_site = case value[:same_site]
+        when true
+          "; SameSite"
+        when Symbol, String
+          "; SameSite=#{value[:same_site]}"
```

Comment:
Similar strictness-loss issue if we whitelist Symbol/String values. Say someone does `same_site: heuristic_to_decide_same_site?` which has a truthy response. Then they'd lose strictness and never know it's because they didn't provide an allowed value. We could case out non-truthy values to ignore them `else` to the `true` case.